kernel/kernel-api/src/main/java/io/delta/kernel/TransactionCommitStatus.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel;
+
+import io.delta.kernel.annotation.Evolving;
+import io.delta.kernel.client.TableClient;
+import io.delta.kernel.utils.CloseableIterable;
+
+/**
+ * Contains the result of a successful transaction commit. Returned by
+ * {@link Transaction#commitWithRetries(TableClient, CloseableIterable, int)}.
+ *
+ * @since 3.2.0
+ */
+@Evolving
+public class TransactionCommitStatus {
```

Comment:
nit: Result? instead of status? Can you poll for the status during the transaction? or do you just get this at the end?

Reply:
the `Transaction.commitWithRetries` is a synchronous call. Changed it to `TransactionCommitResult`.